### PR TITLE
Feat/candy machine handling

### PR DESCRIPTION
**Pull request by d-reader-luka**

### CODE REVIEW

File: src/utils/environment.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const getApiUrl = (environment: string) =>
+  environment === 'dev'
+    ? 'https://api.dev.dreader.io'
+    : 'https://api.dreader.io';
```
Comment (d-reader-josip) [inline src/utils/environment.ts:4]: I'll move this value to .env, and also gonna add the webhook ID

Comment (d-reader-josip) [reply src/utils/environment.ts:4]: We can use the `DOMAIN_NAME` env value for the domain, and `WEBHOOK_ID` for for the webhook.
In the future we will keep track of webhooks in a database table potentially

File: src/vendors/candy-machine.service.ts [hunk 2]
```diff
@@ -296,9 +302,21 @@ export class CandyMachineService {
       iteration = iteration + 1;
     }
 
+    await this.createHeliusWebhook(candyMachine.address.toBase58());
+
```
Comment (d-reader-josip) [inline src/vendors/candy-machine.service.ts:306]: Helius webhook should never be created within the app itself besides on the endpoint for creating it. We can also add a CLI command for it but unnecessary for now. Please remove it from the CandyMachineService constructor.

What we SHOULD do is just keep it updated with new `accountAddresses` whenever we create a candy machine or an NFT

File: src/webhooks/helius/helius.service.ts [hunk 3]
```diff
@@ -72,13 +84,17 @@ export class HeliusService {
...
-      collectionNftAddress: '68gsxWrLkMkwMznXM9qam7FvDK1SDBhVJZaQUpAwXL6k',
-      name: uuidv4(),
+      address: mintAddress.toBase58(),
+      candyMachineAddress: 'FRzbE9ENACT1ag8z4Q1JpQ5chU18GZq26Bxr8Vd71BDb',
+      collectionNftAddress: nft.collection.address.toBase58(),
+      name: nft.name,
       owner: enrichedTransaction.tokenTransfers.at(0).toUserAccount,
-      uri: 'uri',
+      uri: nft.uri,
     };
```
Comment (d-reader-josip) [inline src/webhooks/helius/helius.service.ts:98]: Please consult with Athar on the best approach for getting the NFT name, URI, Collection NFT address, and the Candy Machine address.

If we know the CM we can easily figure out the Collection NFT

File: src/webhooks/helius/helius.service.ts [hunk 4]
```diff
@@ -102,10 +118,32 @@ export class HeliusService {
           itemsMinted: { increment: 1 },
         },
       });
+    const nftTransactionInfo = enrichedTransaction.events.nft as NFTEvent & {
+      amount: number;
+    };
+    const logIntoMintReceipt = this.prisma.mintReceipt.create({
+      data: {
+        buyer: nftTransactionInfo.buyer,
+        price: nftTransactionInfo.amount / LAMPORTS_PER_SOL, // not sure if there is an existing helper function that can convert to SOL properly (max 9 decimals I assume?)
```
Comment (d-reader-josip) [inline src/webhooks/helius/helius.service.ts:127]: Let's remove this comment and store the Lamport value in the database so we don't have to use decimals/floats?

File: src/webhooks/helius/helius.service.ts [hunk 5]
```diff
@@ -102,10 +118,32 @@ export class HeliusService {
...
+        mintedAt: new Date(nftTransactionInfo.timestamp),
```
Comment (d-reader-josip) [inline src/webhooks/helius/helius.service.ts:128]: Let's rename this `mintedAt` column to `timestamp` in the db schema

File: src/webhooks/helius/helius.service.ts [hunk 6]
```diff
@@ -102,10 +118,32 @@ export class HeliusService {
...
-      updateComicIssueCandyMachine,
-    ]);
+    try {
+      const [comicIssueNft] = await this.prisma.$transaction([
+        createComicIssueNft,
+        updateComicIssueCandyMachine,
+        logIntoMintReceipt,
+      ]);
+      const webhook = await this.getMyWebhook(
+        'c2e87057-b3e4-49a1-a84f-a31b0585a69d',
```
Comment (d-reader-josip) [inline src/webhooks/helius/helius.service.ts:140]: use the Webhook ID from the .env

File: src/webhooks/helius/helius.service.ts [hunk 7]
```diff
@@ -102,10 +118,32 @@ export class HeliusService {
...
 
-    await this.prisma.$transaction([
-      createComicIssueNft,
-      updateComicIssueCandyMachine,
-    ]);
+    try {
+      const [comicIssueNft] = await this.prisma.$transaction([
+        createComicIssueNft,
+        updateComicIssueCandyMachine,
+        logIntoMintReceipt,
```
Comment (d-reader-josip) [inline src/webhooks/helius/helius.service.ts:137]: Let's skip using prisma.$transaction for now. If one of these 3 queries fail, it should log an error but we should still try executing remaining queries.

try catch each of these 3 queries and ditch the $transaction

**Review by d-reader-josip — CHANGES_REQUESTED** [review]

File: src/webhooks/helius/helius.service.ts (hunk 3 above)
Comment (d-reader-luka) [reply src/webhooks/helius/helius.service.ts:98]: @athar-d-reader

Comment (athar-d-reader) [reply src/webhooks/helius/helius.service.ts:102]: we can use 
```
    const metadataPda = this.metaplex.nfts().pdas().metadata({mint});
    const info = await this.metaplex.rpc().getAccount(metadataPda);
    const metadata = toMetadata(toMetadataAccount(info));
```
with this we are making only 1 call for the required things and saving us 3 calls to rpc;
we can import ``toMetadata`` and ``toMetadataAccount`` from ``@metaplex-foundation/js``

Comment (d-reader-josip) [reply src/webhooks/helius/helius.service.ts:102]: ty @athar-d-reader 

@d-reader-luka I'll resolve this by Monday and merge the PR in. No need for you to worry about it

**Pull request merged**